modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/endpoint/WorkflowRestService.java

```diff
@@ -361,8 +362,15 @@ public Response getWorkflowsAsXml(@QueryParam("state") List<String> states, @Que
     q.withMediaPackage(mediapackageId);
     q.withCreator(creator);
     q.withContributor(contributor);
-    q.withDateAfter(SolrUtils.parseDate(fromDate));
-    q.withDateBefore(SolrUtils.parseDate(toDate));
+    try {
+      q.withDateAfter(SolrUtils.parseDate(fromDate));
+      q.withDateBefore(SolrUtils.parseDate(toDate));
+    } catch (ParseException e) {
+      return Response
+          .status(Status.BAD_REQUEST)
+          .entity("Invalid date format")
+          .build();
+    }
```

Comment:
@lkiesow should the BAD_REQUEST response be added to the method restResponse declaration?

Reply:
Good catch → fixed